第一种Promise的使用方式

new Promise((resolve,reject)=>{
setTimeout(()=>{
resolve('111'),
reject('222'),
})
}).then(()=>{

console.log('Hellow');
console.log('Hellow');
console.log('Hellow');
console.log('Hellow');

return new Promise((reject,rejective)=>{
setTimeout(()=>{
resolve('111'),
reject('222'),
})
})

}).then(()=>{
console.log('Hellow');
console.log('Hellow');
console.log('Hellow');
console.log('Hellow');

return new Promise((reject,rejective)=>{
setTimeout(()=>{
resolve('111'),
reject('222'),
})
})

}).then(()=>{
console.log('Hellow');
console.log('Hellow');
console.log('Hellow');
console.log('Hellow');
})


第二种Promise的使用方式
wrapperd into（包装）---使用这个指令来包装异步代码
网络请求：aaa -> 自己处理（10行）
处理：aaa111 -> 自己处理（10行）
处理：aaa111222 -> 自己处理（10行）

new Promise ((reject,resolve)=>{
setTimeout(()=>{
resolve("aaa")
},1000)
}).then(res=>{
//1.自己处理相关代码
console.log(res,'第一层的10行代码');

//2.实现res = res + '111'
//reject是可选参数，所以也可以不传入reject的值，所以这里将其省略

<!-- 
return new Promise((resolve)=>{
resolve('aaa')

})


}).then(res=>{
console.log(res,'第一层的10行代码');

return new Promise(resolve =>{
resolve(res+'111')
})

}).then(res =>{
console.log(res,'第二层的10行代码');

return new Promise(resolve => {
    resolve(res+'222')
})

}).then(res=>{
    console.log(res,'第三层的10行代码');

}) -->








在Promise中可以对异步操作进行简写

return new Promise.resolve(res+'111')
    
    
    
<!--     
return new Promise((resolve)=>{
    resolve('aaa')
    
    })
    
    
    }).then(res=>{
    console.log(res,'第一层的10行代码');
    
    return new Promise.resolve(res+'111')
    
    
    }).then(res =>{
    console.log(res,'第二层的10行代码');
    
    return new Promise.resolve(res+'222')
    
    
    }).then(res=>{
        console.log(res,'第三层的10行代码');
    
    })
     -->

    还可以再进一步简写

    直接在return后加上拼接方式或者处理方式    


return new Promise((resolve)=>{
    resolve('aaa')
    
    })
    
    
    }).then(res=>{
    console.log(res,'第一层的10行代码');
    
    return res+'111'
    
    
    }).then(res =>{
    console.log(res,'第二层的10行代码');
    
    return res+'222'
    
    
    }).then(res=>{
        console.log(res,'第三层的10行代码');
    
    })
    
    如果想要查看报错可以使用reject来输出错误信息
    也可以通过 throw来主动抛出报错信息